### PR TITLE
Automatic aspect ratio calculation

### PR DESCRIPTION
**Pull request by eugenioLR**
As I commented on issue #732, I had an idea to calculate the aspect ratio of the figure when using ImageGrid so that there isn't so much white space between subplots when displaying images with uneven aspect ratios.

I implemented it by setting the parameter `aspect` to `'auto'`, which triggers a couple of lines of code to calculate it from the dimensions of the image. Since lists of images are also supported, I also made it so it uses the lowest aspect ratio among the images, since it would make them overlap in the figure otherwise.

### CODE REVIEW

**Comment by eugenioLR** [comment]
It doesn't seem to work on macOS, but it seems like it's not an issue of the pull request since this same thing seems to happen on the other pull requests.

**Review by SarthakJariwala — COMMENTED** [review]
Sorry for the delay. Left a few small comments

**Comment by SarthakJariwala** [comment]
Another thing I would recommend adding are tests for the aspect ratio

**Comment by eugenioLR** [comment]
I saw all the comments and they all seem reasonable. I addressed all the issues you raised, Everything should be correct now.

File: src/seaborn_image/_grid.py [hunk 1]
```diff
@@ -418,6 +431,12 @@ def __init__(
 
             _nimages = len(slices)
 
+            if aspect == "auto":
+                # Select axis where width and height are
+                height_idx, width_idx = [i for i in range(data.ndim) if i != axis][:2]
```
[the remark this answers is not in the document]
Comment (eugenioLR) [reply src/seaborn_image/_grid.py]: I made it so negative axis can work by calculating the modulo with respect to data.ndim.

**Review by SarthakJariwala — APPROVED** [review]
Looks good! Thanks for making the changes.

**Pull request merged**